frontend/server/src/Controllers/Problem.php

```diff
@@ -4662,6 +4667,12 @@ public static function getProblemDetailsForTypeScript(
             'tried' => false,
         ];
 
+        $alreadyReviewedPayload = null;
+
+        if ($alreadyReviewed) {
+            $alreadyReviewedPayload = ['contents' => $reviewedData['contents'], 'qualitynomination_id' => $reviewedData['qualitynomination_id']];
+        }
```

Comment:
En la función `getReviewedData` ya estás regresando la información en el formato que la vas a mandar vía payload. Así que sólo sería necesario validar que no sea `null` para que lo asignes directamente:
```suggestion
        if (!is_null($alreadyReviewed)) {
            $response['templateProperties']['payload']['alreadyReviewedPayload'] = $alreadyReviewed;
        }
```
una vez hecho ese cambio, ya puedes eliminar L4670 y también el bloque de L4724-L4727